app/connectors/ExtraHeaders.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import java.util.UUID.randomUUID
+
+import uk.gov.hmrc.http.HeaderCarrier
+
+trait ExtraHeaders {
+  private def correlationId(hc: HeaderCarrier): String = {
+    val CorrelationIdPattern = """.*([A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}).*""".r
+    hc.requestId match {
+      case Some(requestId) =>
+        requestId.value match {
+          case CorrelationIdPattern(prefix) => prefix + "-" + randomUUID.toString.substring(24)
+          case _                            => randomUUID.toString
+        }
+      case _ => randomUUID.toString
+    }
+  }
+
+  def extraHeaders(implicit hc: HeaderCarrier): HeaderCarrier =
+    hc.withExtraHeaders(
+      "CorrelationId" -> correlationId(hc),
+      "X-Session-ID"  -> hc.sessionId.fold("-")(_.value),
+      "X-Request-ID"  -> hc.requestId.fold("-")(_.value)
```

Comment:
Looks good. I sessionId or requestId is missing it will default to "-". Ok with me. It's common in kibana.